app/models/user.rb

```diff
@@ -2,5 +2,25 @@ class User < ActiveRecord::Base
   # Include default devise modules. Others available are:
   # :confirmable, :lockable, :timeoutable and :omniauthable
   devise :database_authenticatable, :registerable,
-         :recoverable, :rememberable, :trackable, :validatable
+         :recoverable, :rememberable, :trackable, :omniauthable, :validatable
+
+  def self.find_for_google_oauth2(access_token, signed_in_resource=nil)
+    data = access_token.info
+    user = User.where(:provider => access_token.provider, :uid => access_token.uid ).first
```

Comment:
Line is too long. [90/80]<br>Use the new Ruby 1.9 hash syntax.<br>Space inside parentheses detected.